src/Chapter3.hs

```diff
@@ -374,6 +390,20 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Mob = Mob
+  { mobHealth :: Int
+  , mobAttack :: Int
+  , mobGold   :: Int
+  } deriving (Show)
+
+mobFight :: Mob -> Mob -> Int
+mobFight attacker defender
```

Comment:
Nice catch that both of the fighters have the same structure!
The only issue with reusage of the same data type for Monster and Knight in here is that you can easily mix them when calling the function:

```haskell
> mobFight knight monster
> mobFight monster knight
```

Both of the calls are absolutely valid, but the results are going to be different!

Here, we had in mind to make two types for each Monster and Knight. But as you already noticed, these two would be almost identical, and it feels wrong to duplicate that. Later we show the `newtypes` which usually resolve these issues (we didn't expect you to use them in here, this is just the additional note 🙂 )
```haskell
newtype Monster = Monster Mob
newtype Knight = Knight Mob
```